libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -584,7 +584,11 @@ export class BotFrameworkAdapter extends BotAdapter implements ICredentialTokenP
      */
     public async getSignInLink(context: TurnContext, connectionName: string): Promise<string>;
     public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials): Promise<string>;
-    public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials): Promise<string> {
+    public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials, userId?: string, finalRedirect?: string): Promise<string> {
```

Comment:
The new params should be add to at least one of the other signatures.  In TypeScript when you overload a methods signature the last signature is always hidden intellisense wise.

Reply:
@stevenic Will amend. Forgot about that.

Reply:
The other thing you can do is just drop the other two signatures as I don't see that they're adding any value.


---
In reply to: [385957002](https://github.com/microsoft/botbuilder-js/pull/1803#discussion_r385957002) [](ancestors = 385957002)

Reply:
@Stevenic Like this?

```js
public async getSignInLink(context: TurnContext, connectionName: string): Promise<string>;
public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials): Promise<string>;
public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials, userId?: string, finalRedirect?: string): Promise<string>;
public async getSignInLink(context: TurnContext, connectionName: string, oAuthAppCredentials?: AppCredentials, userId?: string, finalRedirect?: string): Promise<string> {
```

Reply:
removed